messaging/src/main/java/org/axonframework/commandhandling/annotation/AnnotationCommandHandlerAdapter.java

```diff
@@ -102,46 +101,47 @@ public AnnotationCommandHandlerAdapter(@Nonnull T annotatedCommandHandler,
                                            @Nonnull ParameterResolverFactory parameterResolverFactory,
                                            @Nonnull HandlerDefinition handlerDefinition,
                                            @Nonnull MessageTypeResolver messageTypeResolver) {
+        this.handlingComponent = SimpleCommandHandlingComponent.create(
+                "AnnotationCommandHandlerAdapter[%s]".formatted(annotatedCommandHandler.getClass().getName())
+        );
         this.target = requireNonNull(annotatedCommandHandler, "The Annotated Command Handler may not be null");
         this.model = AnnotatedHandlerInspector.inspectType((Class<T>) annotatedCommandHandler.getClass(),
                                                            parameterResolverFactory,
                                                            handlerDefinition);
         this.messageTypeResolver = requireNonNull(messageTypeResolver, "The MessageTypeResolver may not be null");
+
+        initializeHandlersBasedOnModel();
     }
 
-    @Override
-    public AnnotationCommandHandlerAdapter<T> subscribe(@Nonnull QualifiedName name,
-                                                        @Nonnull org.axonframework.commandhandling.CommandHandler commandHandler) {
-        throw new UnsupportedOperationException(
-                "This Command Handling Component does not support direct command handler registration."
-        );
+    private void initializeHandlersBasedOnModel() {
+        model.getAllHandlers().forEach(
+                (modelClass, handlers) ->
+                        handlers.stream()
+                                .filter(h -> h.canHandleMessageType(CommandMessage.class))
+                                .forEach(this::registerHandler));
     }
 
-    @Nonnull
-    @Override
-    public MessageStream.Single<CommandResultMessage<?>> handle(@Nonnull CommandMessage<?> command,
-                                                                @Nonnull ProcessingContext processingContext) {
-        MessageHandlingMember<? super T> handler = model.getAllHandlers()
-                                                        .values()
-                                                        .stream()
-                                                        .flatMap(Collection::stream)
-                                                        .filter(ch -> ch.canHandle(command, processingContext))
-                                                        .findFirst()
-                                                        .orElseThrow(() -> new NoHandlerForCommandException(command));
-
-        return model.chainedInterceptor(target.getClass())
-                    .handle(command, processingContext, target, handler)
-                    .mapMessage(this::asCommandResultMessage)
-                    .first()
-                    .cast();
+    private void registerHandler(MessageHandlingMember<? super T> handler) {
```

Comment:
Very nice that you have used non-annotation commandhandler inside. I really like it!

Reply:
Glad you like it :)